ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,170 @@
+    ~doc:"Set the host field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_status =
+  call ~name:"set_status" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the status of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; (Bool, "status", "The status the TracerProvider is to be set to")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_tags =
+  call ~name:"set_tags" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the tags field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Map (String, String)
+        , "tags"
+        , "The tags the TracerProvider is to be set to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_endpoints =
+  call ~name:"set_endpoints" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the endpoints field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set String
+        , "endpoints"
+        , "The endpoints the TracerProvider is to be set to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_components =
+  call ~name:"set_components" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the components field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set String
+        , "components"
+        , "The components the TracerProvider is to be set to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_filters =
+  call ~name:"set_filters" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the filters field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; (Set String, "filters", "The filter the TracerProvider is to be set to")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_processors =
+  call ~name:"set_processors" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the processors field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set String
+        , "processors"
+        , "The processors the TracerProvider is to be set to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let t =
+  create_obj ~name:_tracing ~descr:"Describes a TracerProvider" ~doccomments:[]
+    ~gen_constructor_destructor:false ~gen_events:true ~in_db:true ~lifecycle:[]
+    ~persist:PersistEverything ~in_oss_since:None
+    ~messages_default_allowed_roles:_R_POOL_ADMIN
+    ~contents:
+      ([uid _tracing ~lifecycle:[]]
```

Comment:
Why is the first field special and this is not simply a list?

Reply:
Mistake of copy paste, I have since removed the append and just inlcuded it.

Reply:
Resolved by: [0d0b6af](https://github.com/xapi-project/xen-api/pull/4966/commits/0d0b6aff8e9aaf3ff50907258e3ddd71ad5c0978)